### DISCUSSION TITLE
Using pyinstaller fails

### DISCUSSION THREAD

**Discussion by dmaunder** [tdorssers/TeslaPy#62] [Q&A]
I have a program that is working well in Python 3.10 using teslapy, but when I generate a one file exe, the program compiles without error, but at run time, the  following code fails:

#initialise Tesla API
try:
   with teslapy.Tesla(email=TeslaAccountEmail,timeout=TeslaAccountTimeout) as tesla:
       tesla.fetch_token()
except Exception as e:
   logger.error('Fatal Error initialising Tesla API - ' + str(e))
   exception_logger.exception('Fatal Error initialising Tesla API ' + str(e))
   sys.exit() 

try:
   vehicles = tesla.vehicle_list()
except Exception as e:
   logger.error('Fatal Error in Tesla API getting vehicle list - ' + str(e))
   exception_logger.exception('Fatal Error in Tesla API getting vehicle list - ' + str(e))
   sys.exit() 

When the vehicle_list() call is made, it raises an exception with

 Unknown endpoint name VEHICLE_LIST

Can anyone advise how to resolve this?

**Comment by tdorssers** [comment]
The teslapy folder contains two json files, those probably need to be included using this procedure: https://pyinstaller.readthedocs.io/en/stable/spec-files.html#using-data-files-from-a-module

**Reply by dmaunder** [reply]
Thanks Tim. Sorry, I have read that and have no idea what I need to do with the 2 json files. Can you spell it out for me?

**Comment by tdorssers** [comment]
You need to modify the spec file to include endpoints.json and option_codes.json as data files to the teslapy module. If you don't know the path to these files, then you can some Python code to the spec file and modify the Analysis line:

```python
from os.path import split, join
import teslapy

base_path = split(teslapy.__file__)[0]

a = Analysis(...
     datas=[(join(base_path, 'endpoints.json'), 'teslapy'),
            (join(base_path, 'option_codes.json'), 'teslapy')],
     ...
     )
```
